pkg/aws/services/vpclattice.go

```diff
@@ -224,6 +228,36 @@ func (d *defaultLattice) ListTargetGroupsAsList(ctx context.Context, input *vpcl
 	return result, nil
 }
 
+func (d *defaultLattice) ListTagsForResourceWithContext(ctx context.Context, input *vpclattice.ListTagsForResourceInput, option ...request.Option) (*vpclattice.ListTagsForResourceOutput, error) {
+	key := tagCacheKey(*input.ResourceArn)
+	if d.cache != nil {
+		r, ok := d.cache.Get(key)
+		if ok {
+			return r.(*vpclattice.ListTagsForResourceOutput), nil
+		}
+	}
+	out, err := d.VPCLatticeAPI.ListTagsForResourceWithContext(ctx, input, option...)
+	if err != nil {
+		return nil, err
+	}
+	if d.cache != nil {
+		d.cache.Add(key, out)
+	}
+	return out, nil
+}
+
+func tagCacheKey(arn string) string {
+	return "tag-" + arn
+}
+
+func (d *defaultLattice) TagResourceWithContext(ctx context.Context, input *vpclattice.TagResourceInput, option ...request.Option) (*vpclattice.TagResourceOutput, error) {
+	if d.cache != nil {
```

Comment:
why cache should be nil?

Reply:
I think when using cache we should almost always have a way to switch off for testing purpose, etc.

Reply:
in this case should be interface with no-op cache